src/ServiceFabricPubSub/RequestRouter/Controllers/RequestController.cs

```diff
@@ -21,127 +25,91 @@ public class RequestController : ApiController
 
         private static int _reverseProxyPort;
 
-        // POST api/tenantId/topicName
-        public async Task<HttpResponseMessage> Post(string tenantId, string topicName)
-        {
-            HttpResponseMessage responseMessage = new HttpResponseMessage(HttpStatusCode.InternalServerError);
-
-            // Assuming the message body will contain the content for the data to put to the topic.
-            string messageBody = await this.Request.Content.ReadAsStringAsync();
-
-            HttpServiceUriBuilder builder = new HttpServiceUriBuilder()
+    public async Task<HttpResponseMessage> Post(string tenantId, string topicName, string message)
+    {
+            try
             {
-                PortNumber = await GetReverseProxyPortAsync(),
-                ServiceName = $"{tenantId}/{TenantApplicationTopicServiceName}/{topicName}/api/"
-            };
+                HttpResponseMessage responseMessage = new HttpResponseMessage(HttpStatusCode.OK);
 
-            HttpResponseMessage topicResponseMessage;
-            using (HttpClient httpClient = new HttpClient())
-            {
-                HttpContent postContent = new StringContent(messageBody);
-                postContent.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                var msg = new PubSubMessage() { Message = message };
+                var topicSvc = ServiceProxy.Create<ITopicService>(new Uri($"fabric:/{tenantId}/{TenantApplicationTopicServiceName}/{topicName}"));
+                await topicSvc.Push(msg);
```

Comment:
I had few issues with port conflicts. I can try again to put back the http listener